packages/nimble-components/src/select/index.ts

```diff
@@ -49,51 +146,701 @@ export class Select extends FoundationSelect implements ErrorPattern {
+     *
+     * @internal
+     * @override
+     */
+    public override handleChange(source: unknown, propertyName: string): void {
+        super.handleChange(source, propertyName);
+        if (propertyName === 'value') {
+            this.updateValue();
+        }
+    }
+
+    /**
+     * Prevents focus when size is set and a scrollbar is clicked.
+     *
+     * @param e - the mouse event object
+     *
+     * @override
+     * @internal
+     */
+    public override mousedownHandler(e: MouseEvent): BooleanOrVoid {
+        if (e.offsetX >= 0 && e.offsetX <= this.listbox?.scrollWidth) {
+            return super.mousedownHandler(e);
+        }
+
+        return this.collapsible;
+    }
+
+    /**
+     * Sets the multiple property on the proxy element.
+     *
+     * @param prev - the previous multiple value
+     * @param next - the current multiple value
+     */
+    public override multipleChanged(
+        prev: boolean | undefined,
+        next: boolean
     ): void {
-        if (this.region && this.control) {
-            this.region.anchorElement = this.control;
+        super.multipleChanged(prev, next);
+
+        if (this.proxy) {
+            this.proxy.multiple = next;
         }
     }
 
-    private controlChanged(
-        _prev: HTMLElement | undefined,
-        _next: HTMLElement | undefined
+    public inputClickHandler(e: MouseEvent): void {
+        e.stopPropagation(); // clicking in filter input shouldn't close dropdown
+    }
+
+    /**
+     * Handle content changes on the control input.
+     *
+     * @param e - the input event
+     * @internal
+     */
+    public inputHandler(e: InputEvent): boolean {
+        this.filter = this.filterInputElement?.value ?? '';
+        if (!this.committedSelectedOption) {
+            this.committedSelectedOption = this._options.find(
+                option => option.selected
+            );
+        }
+        this.clearSelection();
+        this.filterOptions();
+
+        if (
+            this.filteredOptions.length > 0
+            && this.committedSelectedOption
+            && !this.filteredOptions.includes(this.committedSelectedOption)
+        ) {
+            const enabledOptions = this.filteredOptions.filter(
+                o => !o.disabled
+            );
+            if (enabledOptions.length > 0) {
+                enabledOptions[0]!.selected = true;
+            } else {
+                // only filtered option is disabled
+                this.selectedOptions = [];
+                this.selectedIndex = -1;
+            }
+        } else if (this.committedSelectedOption) {
+            this.committedSelectedOption.selected = true;
+        }
+
+        if (e.inputType.includes('deleteContent') || !this.filter.length) {
+            return true;
+        }
+
+        e.stopPropagation();
+        return true;
+    }
+
+    public override focusoutHandler(e: FocusEvent): BooleanOrVoid {
+        this.updateSelectedIndexFromFilteredSet();
```

Comment:
@jattasNI @rajsite, everything below this line in this method is copied directly from FAST.